src/client/datascience/interactive-window/interactiveWindowProvider.ts

```diff
@@ -196,18 +201,18 @@ export class InteractiveWindowProvider implements IInteractiveWindowProvider, IA
         // Multiple means use last active window or create a new one
         // if not owned.
         if (interactiveMode === 'multiple') {
-            // Owner being undefined means create a new window, othewise use
+            // Owner being undefined means create a new window, otherwise use
             // the last active window.
             return owner ? this.activeWindow || this.lastActiveInteractiveWindow || this._windows[0] : undefined;
         }
 
         // Otherwise match the owner.
         return this._windows.find((w) => {
-            if (!owner && !w.owner) {
+            if (!owner && !w.owner && !connection) {
                 return true;
             }
             if (owner && w.owner && this.fs.areLocalPathsSame(owner.fsPath, w.owner.fsPath)) {
-                return true;
+                return connection ? w.originalConnection?.id === connection.id : true;
```

Comment:
alternatively:
`!connection || w.originalConnection?.id === connection.id`

Reply:
Personally find this more readable when going with `!connection`, i find reading negations a little difficult.